query/src/sql/statements/statement_alter_table.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use common_exception::ErrorCode;
+use common_exception::Result;
+use common_planners::PlanNode;
+use common_planners::RenameTableEntity;
+use common_planners::RenameTablePlan;
+use common_tracing::tracing;
+use sqlparser::ast::ObjectName;
+
+use crate::sessions::QueryContext;
+use crate::sql::statements::AnalyzableStatement;
+use crate::sql::statements::AnalyzedResult;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct DfAlterTable {
+    pub if_exists: bool,
+    pub table_name: ObjectName,
+    pub action: AlterTableAction,
+
+    // for AlterTableAction
+    pub new_table_name: ObjectName,
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, Copy, Debug, PartialEq)]
+pub enum AlterTableAction {
+    RenameTo,
```

Comment:
```suggestion
    RenameTable(ObjectName),
```

We should put action and data together.